RxCocoa/Common/DelegateProxyType.swift

```diff
@@ -86,56 +86,98 @@ and extend it
 
 */
 public protocol DelegateProxyType : AnyObject {
-    /// DelegateProxy factory
+    associatedtype ParentObject: AnyObject
+    associatedtype Delegate: AnyObject
+    
+    /// DelegateProxy factory. Use 'DelegateProxyFactory.sharedFactory'.
     static var factory: DelegateProxyFactory { get }
 
+    /// Creates new proxy for target object.
+    /// Should not call this function directory, use 'DelegateProxy.proxyForObject'
+    static func createProxy(for object: AnyObject) -> AnyObject
+
+    /// Store DelegateProxy subclass to factory.
+    /// When make 'RxXXXDelegateProxy' subclass, call 'RxXXXDelegateProxySubclass.register()' 1 time.
+    /// 'RxXXXDelegateProxy' can have one subclass implementation per concrete ParentObject type.
+    static func register()
+    
+    /// It is require that enumerate call `register` of the extended DelegateProxy subclasses here.
+    static func knownImplementations()
+
+    init(parentObject: ParentObject)
+
     /// Returns assigned proxy for object.
     ///
     /// - parameter object: Object that can have assigned delegate proxy.
     /// - returns: Assigned delegate proxy or `nil` if no delegate proxy is assigned.
-    static func assignedProxyFor(_ object: AnyObject) -> AnyObject?
-    
+    static func assignedProxyFor(_ object: ParentObject) -> Delegate?
+
     /// Assigns proxy to object.
     ///
     /// - parameter object: Object that can have assigned delegate proxy.
     /// - parameter proxy: Delegate proxy object to assign to `object`.
-    static func assignProxy(_ proxy: AnyObject, toObject object: AnyObject)
-    
+    static func assignProxy(_ proxy: Delegate, toObject object: ParentObject)
+
     /// Returns designated delegate property for object.
     ///
     /// Objects can have multiple delegate properties.
     ///
     /// Each delegate property needs to have it's own type implementing `DelegateProxyType`.
     ///
+    /// It's abstract method.
+    ///
     /// - parameter object: Object that has delegate property.
     /// - returns: Value of delegate property.
-    static func currentDelegateFor(_ object: AnyObject) -> AnyObject?
+    static func currentDelegateFor(_ object: ParentObject) -> Delegate?
```

Comment:
I believe this should now be `func currentDelegate(for object: ParentObject) -> Delegate?`